modules/cli/src/main/scala/scala/cli/commands/Package.scala

```diff
@@ -608,12 +650,64 @@ object Package extends ScalaCommand[PackageOptions] {
     ProcUtil.maybeUpdatePreamble(destPath)
   }
 
+  /** Returns the dependency sub-graph of the provided modules, that is, all their JARs and their
+    * transitive dependencies' JARs.
+    *
+    * Note that this is not exactly the same as resolving those modules on their own (with their
+    * versions): other dependencies in the whole dependency sub-graph may bump versions in the
+    * provided dependencies sub-graph here.
+    *
+    * Here, among the JARs of the whole dependency graph, we pick the ones that were pulled by the
+    * provided modules, and might have been bumped by other modules. This is strictly a subset of
+    * the whole dependency graph.
+    */
+  private def providedFiles(
+    build: Build.Successful,
+    provided: Seq[dependency.AnyModule],
+    logger: Logger
+  ): Either[BuildException, Seq[os.Path]] = either {
+
+    logger.debug(s"${provided.length} provided dependencies")
+    val res = build.artifacts.resolution.getOrElse {
+      sys.error("Internal error: expected resolution to have been kept")
+    }
+    val modules = value {
+      provided
+        .map(_.toCs(build.scalaParams))
+        .sequence
+        .left.map(CompositeBuildException(_))
+    }
+    val modulesSet   = modules.toSet
+    val providedDeps = res.dependencyArtifacts.map(_._1).filter(dep => modulesSet(dep.module))
```

Comment:
```suggestion
    val providedDeps = res.dependencyArtifacts.map(_._1).filter(dep => modulesSet.contains(dep.module))
```

`apply` method on Set is a bit unintuitive

Reply:
Agreed, should be fixed.